src/state/slices/workerpool_slice.ts

```diff
@@ -0,0 +1,11 @@
+import { StateCreator } from "zustand";
+
+import SharedWorkerPool from "../../colorizer/workers/SharedWorkerPool";
+
+export type WorkerPoolSlice = {
+  workerPool: SharedWorkerPool;
+};
+
+export const createWorkerPoolSlice: StateCreator<WorkerPoolSlice, [], [], WorkerPoolSlice> = (_set, _get) => ({
+  workerPool: new SharedWorkerPool(),
```

Comment:
I'm totally confused as to how this is "state".  
Is this going here because it's a shared module that a lot of other components need to access?  
How do you decide what to put in state slices?

Reply:
to offer an alternative maybe there could just be a single global sharedworkerpool instance that lives in the workerpool module itself.  You might have to control how/when it's initialized, of course, but I think it doesn't really depend on anything else at creation time...  Just a thought, I might be totally of base :)